### ISSUE TITLE
Cameras with RenderTarget::Image freeze when their Image is mutably accessed

### ISSUE THREAD

**Issue by uber5001** [bevyengine/bevy#6480]
## Bevy version

0.8.1 and main branch on commit ea4aeff9ec8e204b8d4bc8cbb10c28e90fe271ff

## \[Optional\] Relevant system information

```ignore
AdapterInfo { name: "NVIDIA GeForce RTX 3080 Ti", vendor: 4318, device: 8712, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "512.15", backend: Vulkan }
```

## What you did

I modified the example [post_processing.rs](https://github.com/bevyengine/bevy/blob/ea4aeff9ec8e204b8d4bc8cbb10c28e90fe271ff/examples/shader/post_processing.rs) by adding this system:

```rust
fn break_all_render_textures_system(mut images: ResMut<Assets<Image>>) {
    for _ in images.iter_mut() {}
}
```

Accessing the images with `images.get_mut` will have the same effect.

## What went wrong

I was expecting to see a spinning cube with chromatic aberration. Instead, I see a *still* cube with chromatic aberration. Bevy is only rendering the first frame and then freezing.

Expected:
![Recording 2022-11-05 at 04 30 43](https://user-images.githubusercontent.com/3348363/200115685-1db5cb1f-346a-4fed-90fb-ecd46069f2af.gif)

Actual: 
![Recording 2022-11-05 at 04 29 20](https://user-images.githubusercontent.com/3348363/200115700-8060b710-8f57-4e2d-ae8e-aedc7c9c0923.gif)

## Additional information

On 0.8.1, when using this mutable access to resize the image, I needed to use a workaround from #5595 (which seems to be fixed on main right now).

It seems like a stale render texture is being used. If the Image is resized, the shown frozen image is still at the old size.

**Comment by alice-i-cecile** [comment]
Can you reproduce this on main? :)

**Comment by uber5001** [comment]
Yep, I can reproduce it on main, on commit 5ae94750a199693983c833838729d036fab4f95c

**Comment by alice-i-cecile** [comment]
Thanks for checking :)

**Comment by FishHawk** [comment]
Same in 0.11.0.

**Comment by ostwilkens** [comment]
Experiencing this in 0.13
